tajo-catalog/tajo-catalog-server/src/main/java/org/apache/tajo/catalog/store/AbstractDBStore.java

```diff
@@ -2122,8 +2122,10 @@ private void setPartitionKeys(int pid, PartitionDescProto.Builder partitionDesc)
     List<TablePartitionProto> partitions = new ArrayList<TablePartitionProto>();
 
     try {
-      String sql = "SELECT " + COL_PARTITIONS_PK + ", " + COL_TABLES_PK + ", PARTITION_NAME, " +
-        " PATH FROM " + TB_PARTTIONS;
+      String sql = "SELECT A." + COL_PARTITIONS_PK + ", A." + COL_TABLES_PK + ", A.PARTITION_NAME, " +
+        " A.PATH FROM " + TB_PARTTIONS + " A JOIN " + TB_TABLES + " B "
```

Comment:
Since it is guaranteed that the tables corresponding to partitions always exist, join looks not necessary.